hs_restclient/__init__.py

```diff
@@ -477,16 +482,19 @@ def getResource(self, pid, destination=None, unzip=False):
         :param unzip: True if the bag should be unzipped when saved to destination. Bag contents to be saved to
             directory named $(PID) residing in destination. Only applies when destination is not None.
 
+        :param wait_for_bag_creation: True if to wait to download the bag in case the bag is not ready
+            (bag needs to be recreated before it can be downloaded).
         :raises: HydroShareArgumentException if any arguments are invalid.
         :raises: HydroShareNotAuthorized if the user is not authorized to access the
             resource.
         :raises: HydroShareNotFound if the resource was not found.
         :raises: HydroShareHTTPException to signal an HTTP error
+        :raise: HydroShareBagNotReady if the bag is not reday to be downloaded and wait_for_bag_download is False
```

Comment:
@hyi good catch on parameter name wrong in the documentation. The exception should be raised if the user says not to wait and we find the bag is not ready.